File: Messengerv2/app/src/main/java/com/magnet/magntetchatapp/ui/activities/section/login/LoginActivity.java
package com.magnet.magntetchatapp.ui.activities.section.login;

import android.support.annotation.NonNull;

import com.magnet.magntetchatapp.R;
import com.magnet.magntetchatapp.mvp.api.LoginContract;
import com.magnet.magntetchatapp.ui.activities.abs.BaseActivity;
import com.magnet.magntetchatapp.ui.activities.section.edit.EditProfileActivity;
import com.magnet.magntetchatapp.ui.activities.section.register.RegisterActivity;
import com.magnet.magntetchatapp.ui.views.section.login.DefaultLoginView;

import butterknife.InjectView;

public class LoginActivity extends BaseActivity {

    @InjectView(R.id.viewLogin)
    DefaultLoginView loginView;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected int getMenuId() {
        return NONE_MENU;
    }

    @Override
    protected void onCreateActivity() {
        loginView.setLoginActionCallback(loginActionCallback);
        loginView.onCreateActivity();
    }

    @Override
    protected void onResume() {
        super.onResume();
        loginView.onResumeActivity();
    }

    @Override
    protected void onPause() {
        super.onPause();
        loginView.onPauseActivity();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        loginView.onDestroyActivity();
    }

    /**
     * Callback which provide the action when some actions happened i the login view
     */
    private final LoginContract.OnLoginActionCallback loginActionCallback = new LoginContract.OnLoginActionCallback() {
        @Override
        public void onLoginSuccess() {
            startActivityWithClearTop(EditProfileActivity.class);
        }

        @Override
        public void onLoginError(@NonNull String errorMessage) {
            showMessage(errorMessage);
        }

        @Override
        public void onRegisterPressed() {
            startActivity(RegisterActivity.class, false);
        }
    };
}

File: KMessenger/app/src/main/kotlin/com/magnet/kmessenger/ui/LoginActivity.kt
package com.magnet.kmessenger.ui

import android.os.Bundle
import com.magnet.kmessenger.R
import com.magnet.magnetchat.presenters.LoginContract
import com.magnet.magnetchat.ui.activities.MMXBaseActivity
import com.magnet.magnetchat.ui.views.login.AbstractLoginView

/**
 * Created by aorehov on 26.05.16.
 */

class LoginActivity : MMXBaseActivity() {

    val uiLogin: AbstractLoginView by lazy { findViewById(R.id.login) as AbstractLoginView }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        uiLogin.loginActionCallback = loginCallback
        uiLogin.onCreateActivity()
    }

    override fun onPause() {
        uiLogin.onPauseActivity()
        super.onPause()
    }

    override fun onResume() {
        super.onResume()
        uiLogin.onResumeActivity()
    }

    override fun onDestroy() {
        uiLogin.onDestroyActivity()
        super.onDestroy()
    }

    override fun getLayoutResource(): Int = R.layout.activity_login

    val loginCallback = object : LoginContract.OnLoginActionCallback {
        override fun onLoginSuccess() = startActivity(HomeActivity::class.java)

        override fun onLoginError(errorMessage: String) = showMessage(errorMessage)

        override fun onRegisterPressed() = startActivity(RegisterActivity::class.java)
    }

}